test/unit/app/controllers/metametrics-test.js

```diff
@@ -0,0 +1,206 @@
+    isDevOrTestEnvironment: true,
+    segmentMock: segment,
+    segmentLegacyMock: segmentLegacy,
+    getCurrentChainId: () => chainId,
+    getProviderConfig: () => providerConfig,
+    flushAt: 2,
+    flushInterval: 1000,
+    preferencesStore: {
+      getState() {
+        return {
+          participateInMetaMetrics,
+          metaMetricsId,
+          currentLocale: 'en_US',
+        }
+      },
+    },
+    version: '0.0.1',
+    environment: 'test',
+  })
+}
+describe('MetaMetricsController', function () {
+  describe('constructor', function () {
+    it('should properly initialize', function () {
+      const metaMetricsController = getMetaMetricsController()
+      assert.strictEqual(metaMetricsController.version, '0.0.1-test')
+      assert.strictEqual(metaMetricsController.network, 'Mainnet')
+      assert.strictEqual(metaMetricsController.chainId, '0x1')
+      assert.strictEqual(metaMetricsController.participateInMetaMetrics, true)
+      assert.strictEqual(metaMetricsController.metaMetricsId, '0xabc')
+      assert.strictEqual(metaMetricsController.locale, 'en-US')
+    })
+  })
+
+  describe('trackEvent', function () {
+    it('should track a legacy event', function () {
+      const mock = sinon.mock(segmentLegacy)
+      const metaMetricsController = getMetaMetricsController()
+      mock
+        .expects('track')
+        .once()
+        .withArgs({
+          event: 'Fake Event',
+          userId: TEST_META_METRICS_ID,
+          context: DEFAULT_TEST_CONTEXT,
+          properties: {
+            test: 1,
+            ...DEFAULT_PROPERTIES,
+          },
+        })
+      metaMetricsController.trackEvent(
+        {
+          event: 'Fake Event',
+          category: 'Unit Test',
+          properties: {
+            test: 1,
+          },
+        },
+        { matomoEvent: true },
+      )
+      mock.verify()
+      mock.restore()
+    })
+
+    it('should track a non legacy event', function () {
+      const mock = sinon.mock(segment)
+      const metaMetricsController = getMetaMetricsController()
+      mock
+        .expects('track')
+        .once()
+        .withArgs({
+          event: 'Fake Event',
+          userId: TEST_META_METRICS_ID,
+          context: DEFAULT_TEST_CONTEXT,
+          properties: {
+            test: 1,
+            ...DEFAULT_PROPERTIES,
+          },
+        })
+      metaMetricsController.trackEvent({
+        event: 'Fake Event',
+        category: 'Unit Test',
+        properties: {
+          test: 1,
+        },
+      })
+      mock.verify()
+      mock.restore()
+    })
+
+    it('should immediately flush queue if flushImmediately set to true', async function () {
+      const metaMetricsController = getMetaMetricsController()
+      const value = await metaMetricsController.trackEvent(
+        {
+          event: 'Fake Event',
+          category: 'Unit Test',
+        },
+        { flushImmediately: true },
+      )
+      assert.strictEqual(value, undefined)
```

Comment:
wasn't sure how to test that a promise resolves in < certain ms, so this could be improved.

Reply:
I created a utility to help with this, in `test/lib/wait-until-called.js`. You can see how it's used in the incoming transaction controller tests.